examples/many-clients.rs

```diff
@@ -20,18 +20,18 @@ fn handler(c: Chan<(), Server>) {
 fn server(rx: Receiver<Chan<(), Server>>) {
     let mut count = 0;
     loop {
-        match borrow_request(&rx) {
-            Some(c) => {
-                spawn(move || handler(c));
+        match rx.recv() {
+            Ok(c) => {
+                spawn(move || server_handler(c));
```

Comment:
It seems like I can remove this particular annotation, but not the one in the `main` function. Any idea why?

The error I'm getting is this:

```
examples/many-clients.rs:56:9: 56:14 error: the trait `core::marker::Sync` is not implemented for the type `core::cell::UnsafeCell<std::sync::mpsc::Flavor<session_types::Chan<(), session_types::Recv<u8, session_types::Choose<session_types::Send<u8, session_types::Eps>, session_types::Eps>>>>>` [E0277]
examples/many-clients.rs:56         spawn(|| {
                                    ^~~~~
examples/many-clients.rs:56:9: 56:14 note: `core::cell::UnsafeCell<std::sync::mpsc::Flavor<session_types::Chan<(), session_types::Recv<u8, session_types::Choose<session_types::Send<u8, session_types::Eps>, session_types::Eps>>>>>` cannot be shared between threads safely
examples/many-clients.rs:56         spawn(|| {
                                    ^~~~~
error: aborting due to previous error
```